### DISCUSSION TITLE
HTTP_X_REAL_IP correct with this NPM implementation ?

### DISCUSSION THREAD

**Discussion by MedShake** [jlesage/docker-nginx-proxy-manager#300] [Q&A]
Hi ! 
Thanks for your work ! 
Someone can tell me if this implementation of NPM suffers from bad client IP forwarding ? 
I use the official NPM image and it seems to be impossible to get the original client IP in the final host. 
Personally, I always get my gateway IP as HTTP_X_REAL_IP and it's relay annoying for fail2ban.  
Thanks for your answer !

B.

**Comment by xeviff** [comment]
Hello. I have the same concern.
In my case is a serious issue, because all external traffic (hackers included) comes from the same local IP (the NPM docker one), like 192.168.80.2 and then, when I receive a brute force attack, my NAS blocks the IP, leaving  myself and all my users unable to access anymore to any of my services (all DNS requests are blocked). I was about to open an issue but then I found this topic
